test/util/utils.go

```diff
@@ -0,0 +1,394 @@
+// Copyright 2014-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/ec2"
+	"github.com/aws/amazon-ecs-agent/agent/handlers"
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/ecs"
+	"github.com/fsouza/go-dockerclient"
+)
+
+var ECS *ecs.ECS
+
+func init() {
+	if iid, err := ec2.GetInstanceIdentityDocument(); err == nil {
+		ECS = ecs.New(&aws.Config{Region: iid.Region})
+	} else {
+		ECS = ecs.New(nil)
+	}
+}
+
+// GetTaskDefinition is a helper that provies the family:revision for the named
+// task definition where the name matches the folder in which the task
+// definition is present
+func GetTaskDefinition(name string) (string, error) {
+	_, filename, _, _ := runtime.Caller(0)
+	tdData, err := ioutil.ReadFile(filepath.Join(path.Dir(filename), "..", "testdata", "taskdefinitions", name, "task-definition.json"))
+	if err != nil {
+		return "", err
+	}
+
+	registerRequest := &ecs.RegisterTaskDefinitionInput{}
+	err = json.Unmarshal(tdData, registerRequest)
+	if err != nil {
+		return "", err
+	}
+
+	tdHash := fmt.Sprintf("%x", md5.Sum(tdData))
```

Comment:
Can you add an explanatory comment for future maintainers here?  I understand (and appreciate) that you're creating an hash of the Task Definition and using it as an idempotency token, but that's not really that obvious from your code.